src/corner-dialog/src/CornerDialog.js

```diff
@@ -134,6 +137,16 @@ export default class CornerDialog extends PureComponent {
      */
     containerProps: PropTypes.object,
 
+    /**
+     * Props that are passed to the dialog content body.
+     */
+    contentProps: PropTypes.object,
+
+    /**
+     * Props that are passed to the dialog header.
+     */
+    headerProps: PropTypes.object,
```

Comment:
We may want to avoid creating more of these types of pass-through props to spread on children because it creates a tightly-coupled dependency on how consumers are using it – there's no way of knowing when you are making a breaking change that might impact those props.

In general I'd prefer greater control over the wrapper component's API with _less customization_ and when more customization is needed, we should export the child components, something like:

```js
import { CornerDialog } from 'evergreen-ui'

const MyCornerDialog = ({ title }) => (
  <CornerDialog isShown>
    <CornerDialog.Header {...headerStyles}>{title}</CornerDialog.Header>
    The content
  </CornerDialog>
)
```

Reply:
I see, makes sense!

Can do!

Reply:
@mshwery Would we be able to change this pr to add only the `hasHeader` prop so it matches parity w/ the modals?